### PR TITLE
Make Electrolytic Breathing Unit fill any main hand item that can hold hydrogen

### PR DESCRIPTION
**Pull request by ToMe25**
## Changes proposed in this pull request:
 * Fix Electrolytic Breathing Unit not working when standing on a trapdoor in two block deep water
 * Make Electrolytic Breathing Unit fill any chestplate item that can hold hydrogen
 * Allow Electrolytic Breathing Unit to fill the item in the players main hand with hydrogen

This does not fix the issue with sprinting underwater.
This implements mekanism/Mekanism-Feature-Requests#45

### CODE REVIEW

File: src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java [hunk 1]
```diff
@@ -45,15 +44,21 @@
...
                 GasStack hydrogenStack = new GasStack(MekanismGases.HYDROGEN.get(), maxRate * 2);
                 ItemStack chestStack = player.getItemStackFromSlot(EquipmentSlotType.CHEST);
-                Optional<IGasHandler> capability = MekanismUtils.toOptional(chestStack.getCapability(Capabilities.GAS_HANDLER_CAPABILITY));
-                if (Modules.load(chestStack, Modules.JETPACK_UNIT) != null && capability.isPresent()) {
-                    hydrogenUsed = maxRate * 2 - capability.get().insertChemical(hydrogenStack, Action.EXECUTE).getAmount();
+                Optional<IGasHandler> chestCapability = MekanismUtils.toOptional(chestStack.getCapability(Capabilities.GAS_HANDLER_CAPABILITY));
+                if (chestCapability.isPresent()) {
+                    hydrogenUsed = maxRate * 2 - chestCapability.get().insertChemical(hydrogenStack, Action.EXECUTE).getAmount();
+                }
+                hydrogenStack.setAmount(maxRate * 2 - hydrogenUsed);
```
Comment (pupnewfster) [inline src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java:57]: Moving this in the above if statement and changing it to `hydrogenStack.shrinkStack(hydrogenUsed)` may make this easier to see what is going on

File: src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java [hunk 2]
```diff
@@ -45,15 +44,21 @@
...
+                ItemStack handStack = player.getItemStackFromSlot(EquipmentSlotType.MAINHAND);
+                Optional<IGasHandler> handCapability = MekanismUtils.toOptional(handStack.getCapability(Capabilities.GAS_HANDLER_CAPABILITY));
+                if (handCapability.isPresent()) {
+                	hydrogenUsed -= handCapability.get().insertChemical(hydrogenStack, Action.EXECUTE).getAmount();
```
Comment (pupnewfster) [inline src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java:61]: Given that `hydrogenUsed` is total hydrogen used, and we adjust the hydrogen stack to be what the amount we have is, and that `insertChemical` returns the remainder, shouldn't this line be basically identical to the one for the chestCapability. That way it is `max - remainingAmount`

Comment (ToMe25) [reply src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java:61]: insertChemical returns the remainder? I just assumed it returns the inserted amount.
In that case, yes, it should be max - remainder.

Comment (pupnewfster) [reply src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java:61]: Ya, our handlers (including internal implementation of fluid handling) are closer in behavior to forge's item handlers than forge's fluid handlers in that we have insert return the remainder (as it helps cut down on a decent bit of code across the board for doing repetitive functions)

Comment (ToMe25) [reply src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java]: done.

**Pull request merged**